dropwizard-hibernate/src/main/java/io/dropwizard/hibernate/UnitOfWorkApplicationListener.java

```diff
@@ -75,7 +75,7 @@ public void onEvent(RequestEvent event) {
                 UnitOfWork unitOfWork = methodMap.get(event.getUriInfo()
                         .getMatchedResourceMethod().getInvocable().getDefinitionMethod());
                 unitOfWorkAspect.beforeStart(unitOfWork);
-            } else if (event.getType() == RequestEvent.Type.RESP_FILTERS_START) {
+            } else if (event.getType() == RequestEvent.Type.FINISHED) {
```

Comment:
Hi, I think this may have introduced a bug. Now the session will be closed and the transaction committed only after the serialization has happened. So if there are any database exceptions during the transaction commit (column val exceed/constraints etc..) they happen only after the serialized response has been sent back. So the client assumes things are fine, but actually the server logs show the exception.

I guess we will still need to end the transaction or commit it in the RESP_FILTERS_START event. And session can be closed in FINISHED event.